src/waveform/renderers/allshader/waveformrenderbeat.h

```diff
@@ -14,8 +14,10 @@ class WaveformRenderBeat;
 } // namespace allshader
 
 class allshader::WaveformRenderBeat final
-        : public ::WaveformRendererAbstract,
+        : public QObject,
+          public ::WaveformRendererAbstract,
           public rendergraph::GeometryNode {
+    Q_OBJECT
   public:
     explicit WaveformRenderBeat(WaveformWidgetRenderer* waveformWidget,
             ::WaveformRendererAbstract::PositionSource type =
```

Comment:
I don't get why this needs to be a Q_OBJECT. You don't need to declare a member function as slot in order to bind it to a signal. Any member function is automatically a slot. Only the object that emits the signal needs to be the QObject afaik.

Reply:
It's not a warning, it prevents compiling entirely. Or is this a "warning as error" case?

> is not related to any other changes in the class?

Kinda  - this class was never build with `allshader=allshader_sg` and `rendergraph=rendergraph_sg` before so these are the "major changes" here

Reply:
Ah I see. Yeah makes sense. Thanks for the explanation.